### PR TITLE
dnnl matmul tensor dimension check

### PR DESCRIPTION
**Pull request by jeyblu**
**Description**: check that input tensor dimensions are 2 or higher for dnnl matmul

**Motivation and Context**
input tensor dimensions of 0 and 1 not supported

### CODE REVIEW

File: onnxruntime/core/providers/dnnl/subgraph/dnnl_matmul.h [hunk 1]
```diff
@@ -76,6 +76,7 @@ class DnnlMatmul : public DnnlKernel {
     TensorShape w_shape(wshape, wdim);
 
     AdjustSrcWeightsShape(x_shape, w_shape);
+    ort_source_format_ = GetSourceFormat(static_cast<int>(x_shape.NumDimensions()));
```
Comment (jywu-msft) [inline onnxruntime/core/providers/dnnl/subgraph/dnnl_matmul.h:79]: is it okay we overwrite the values calculated in lines 58 and 65 in the if/else block above.

Comment (jeyblu) [reply onnxruntime/core/providers/dnnl/subgraph/dnnl_matmul.h]: Removed lines 78 and 79:
```
    AdjustSrcWeightsShape(x_shape, w_shape);
    ort_source_format_ = GetSourceFormat(static_cast<int>(x_shape.NumDimensions()));
```

dnnl requires input tensors to have the same dimension. Tried to work around it by using `AdjustSrcWeightsShape(x_shape, w_shape)` to pad the dimension of the tensor whose dimension is smaller to match the dimension of the other tensor. This is now checked in the function `bool IsDimensionSupported(const Node* node) const` in dnnl_exection_provider.h to only allow matmul nodes whose input tensors dimensions are equal.

Something to consider: would it be possible for onnxruntime framework to check the tensor dimension of matmul nodes and if they are not equal, to pad them beforehand? Thanks.

**Comment by jywu-msft** [comment]
can you merge master? I need to re-start the failing CI.

**Comment by jeyblu** [comment]
done. rebased to master

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**